### PR TITLE
PP-6267 Abstract queue functions and common fields

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT
- Move SQS queue specific functionality to abstract class which makes it easier to replace with different queue technology if needed and reuse functionality if new queues are added
- Also extracts common fields to parent class

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/queue/AbstractQueue.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package uk.gov.pay.connector.queue;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import uk.gov.pay.connector.queue.sqs.SqsQueueService;
+
+import java.util.List;
+
+public abstract class AbstractQueue {
```
Comment (sfount) [inline src/main/java/uk/gov/pay/connector/queue/AbstractQueue.java:8]: Does this need to be called `AbstractQueue` if it has the `abstract` property? `CaptureQueue extends Queue` should be understandable

File: src/main/java/uk/gov/pay/connector/queue/CaptureQueue.java [hunk 2]
```diff
@@ -15,42 +15,32 @@
...
+                        .getFailedCaptureRetryDelayInSeconds());
     }
 
     public void sendForCapture(ChargeEntity charge) throws QueueException {
         String message = new GsonBuilder()
                 .create()
                 .toJson(ImmutableMap.of("chargeId", charge.getExternalId()));
 
-        QueueMessage queueMessage = sqsQueueService.sendMessage(captureQueueUrl, message);
+        QueueMessage queueMessage = sendMessageToQueue(message);
```
Comment (sfount) [inline src/main/java/uk/gov/pay/connector/queue/CaptureQueue.java:37]: nice 👍

**Review by sfount — APPROVED** [review]
Really nice clean up 👍

File: src/main/java/uk/gov/pay/connector/queue/AbstractQueue.java (hunk 1 above)
Comment (kbottla) [reply src/main/java/uk/gov/pay/connector/queue/AbstractQueue.java:8]: Think we have mixed conventions in this area and either way seems fine. Java core API has lots of classes starting with Abstract* https://docs.oracle.com/javase/6/docs/api/ which this is based on.

**Pull request merged**